### PR TITLE
 [VCX] enable generic connection messages, signing and verification

### PR DESCRIPTION
**Pull request by glowkey**

### CODE REVIEW

**Comment by jovfer** [comment]
@Artemkaaas please review

**Comment by glowkey** [comment]
I'm sorry, I don't understand. I had to change it from an enum to strings because the message type can be anything, not just one of the enum values.

File: vcx/libvcx/src/utils/logger.rs [hunk 1]
```diff
@@ -140,7 +140,7 @@ impl LibvcxDefaultLogger {
 
         // ensures that the test that is calling this wont fail simply because
         // the user did not set the RUST_LOG env var.
-        let pattern = Some(env::var("RUST_LOG").unwrap_or("debug".to_string()));
+        let pattern = Some(env::var("RUST_LOG").unwrap_or("trace".to_string()));
```
Comment (hadleym) [inline vcx/libvcx/src/utils/logger.rs]: Trace default?

File: vcx/libvcx/src/credential_def.rs [hunk 2]
```diff
@@ -18,7 +18,7 @@ pub struct CredentialDef {
     tag: String,
     name: String,
     source_id: String,
-    issuer_did: String,
+    issuer_did: Option<String>,
```
Comment (hadleym) [inline vcx/libvcx/src/credential_def.rs]: I'm confused on how this is an optional parameter.  I guess its no different than when we had it originally be an empty string upon creation.  Perhaps in a future iteration we require it as a parameter to create?

Comment (glowkey) [reply vcx/libvcx/src/credential_def.rs]: Yes, this is optional. It is not required in the code and must be optional in order to not break compatibility.

File: vcx/libvcx/src/utils/logger.rs (hunk 1 above)
Comment (glowkey) [reply vcx/libvcx/src/utils/logger.rs]: Yes, most of the vcx statements are trace so that should be the default.

**Pull request merged**